src/jit/lir.cpp

```diff
@@ -1110,6 +1113,28 @@ bool LIR::Range::CheckLIR(Compiler* compiler, bool checkUnusedValues) const
     // Make sure the range itself is valid.
     assert(IsValid());
 
+    // Make sure the list is well-formed: no node appears more than once (no circularities). Insert each node in a set
```

Comment:
As mentioned below, SeqTree should never be called on a tree that points to existing LIR. Unfortunately I'm not quite sure how to write an assert that catches that :(